File: 1460/solution.ts
function canBeEqual(target: number[], arr: number[]): boolean {
  target.sort((a, b) => a - b);
  arr.sort((a, b) => a - b);
  for (let i = 0; i < arr.length; i++) {
    if (target[i] !== arr[i]) {
      return false;
    }
  }
  return true;
}

File: 1460/solution.java
class Solution {
    public boolean canBeEqual(int[] target, int[] arr) {
        int[] counts = new int[1001];
        for (int num : target) {
            counts[num]++;
        }
        for (int num : arr) {
            counts[num]--;
        }
        for (int num : counts) {
            if (num != 0) {
                return false;
            }
        }
        return true;
    }
}

// time O(n)
// space O(n)